### DISCUSSION TITLE
Mac is Invalid - Upgrade

### DISCUSSION THREAD

**Discussion by lnahuelituom** [firefly-iii/firefly-iii#5413] [Firefly III]
Hola!

Recientemente actualice a la version 5.6.4 correctamente.

Pero al querere avanzar a las siguientes 5.6.5 y hasta 8 inclusive, recibo el error "mac is invalid" cuando ejecuto php artisan firefly-iii:upgrade-database

Alguna ayuda que pueda darme?

 Hi there,

Firefly III v5.6.8 ran into an error: The MAC is invalid..

The error was of type "Illuminate\Contracts\Encryption\DecryptException".

The error occurred on/at: Tue, 21 Dec 2021 08:31:27 -0300.

This error occurred in file "/var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Encryption/Encrypter.php" on line 227 with code 0.

There was no user logged in for this error or no user was detected.

The IP address related to this error is: 127.0.0.1 (info)
URL is: http://localhost
User agent: Symfony

The full stacktrace is below. If you think this is a bug in Firefly III, you can forward this message to james@firefly-iii.org. This can help fix the bug you just encountered.

If you prefer, you can also open a new issue on GitHub.

The full stacktrace is below:

#0 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Encryption/Encrypter.php(158): Illuminate\Encryption\Encrypter->getJsonPayload()

#1 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Illuminate\Encryption\Encrypter->decrypt()
#2 /var/www/html/firefly-iii-updated/app/Support/System/OAuthKeys.php(126): Illuminate\Support\Facades\Facade::__callStatic()
#3 /var/www/html/firefly-iii-updated/app/Console/Commands/Integrity/RestoreOAuthKeys.php(129): FireflyIII\Support\System\OAuthKeys::restoreKeysFromDB()
#4 /var/www/html/firefly-iii-updated/app/Console/Commands/Integrity/RestoreOAuthKeys.php(77): FireflyIII\Console\Commands\Integrity\RestoreOAuthKeys->restoreKeysFromDB()
#5 /var/www/html/firefly-iii-updated/app/Console/Commands/Integrity/RestoreOAuthKeys.php(56): FireflyIII\Console\Commands\Integrity\RestoreOAuthKeys->restoreOAuthKeys()
#6 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): FireflyIII\Console\Commands\Integrity\RestoreOAuthKeys->handle()
#7 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#8 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#9 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod()
#10 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\Container\BoundMethod::call()
#11 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\Container\Container->call()
#12 /var/www/html/firefly-iii-updated/vendor/symfony/console/Command/Command.php(298): Illuminate\Console\Command->execute()
#13 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\Component\Console\Command\Command->run()
#14 /var/www/html/firefly-iii-updated/vendor/symfony/console/Application.php(1005): Illuminate\Console\Command->run()
#15 /var/www/html/firefly-iii-updated/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#16 /var/www/html/firefly-iii-updated/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun()
#17 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Console/Application.php(94): Symfony\Component\Console\Application->run()
#18 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Console/Application.php(186): Illuminate\Console\Application->run()
#19 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(263): Illuminate\Console\Application->call()
#20 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Illuminate\Foundation\Console\Kernel->call()
#21 /var/www/html/firefly-iii-updated/app/Console/Commands/Upgrade/UpgradeDatabase.php(117): Illuminate\Support\Facades\Facade::__callStatic()
#22 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): FireflyIII\Console\Commands\Upgrade\UpgradeDatabase->handle()
#23 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/Util.php(40): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#24 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#25 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\Container\BoundMethod::callBoundMethod()
#26 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Container/Container.php(653): Illuminate\Container\BoundMethod::call()
#27 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Console/Command.php(136): Illuminate\Container\Container->call()
#28 /var/www/html/firefly-iii-updated/vendor/symfony/console/Command/Command.php(298): Illuminate\Console\Command->execute()
#29 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Console/Command.php(121): Symfony\Component\Console\Command\Command->run()
#30 /var/www/html/firefly-iii-updated/vendor/symfony/console/Application.php(1005): Illuminate\Console\Command->run()
#31 /var/www/html/firefly-iii-updated/vendor/symfony/console/Application.php(299): Symfony\Component\Console\Application->doRunCommand()
#32 /var/www/html/firefly-iii-updated/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun()
#33 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Console/Application.php(94): Symfony\Component\Console\Application->run()
#34 /var/www/html/firefly-iii-updated/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129): Illuminate\Console\Application->run()
#35 /var/www/html/firefly-iii-updated/artisan(37): Illuminate\Foundation\Console\Kernel->handle()
#36 {main}

**Comment by JC5** [comment]
English please thank you

**Comment by lnahuelituom** [comment]
Hello!

Recently update to version 5.6.4 successfully.

But when I want to advance to the following 5.6.5 and up to 8 inclusive, I get the error "mac is invalid" when I run php artisan firefly-iii: upgrade-database

Any help you can give me?

**Reply by JC5** [reply]
Thanks, yes I think so! Can you access the database directly, by any chance?

**Reply by lnahuelituom** [reply]
Yes I think so.

Maybe you should look technically how to do it. But I have access to the server with permissions.

Thank you

**Reply by JC5** [reply]
You must delete two entries in the table `configuration`. The ones called "oauth_private_key" and "oauth_public_key".

**Reply by lnahuelituom** [reply]
that worked perfectly!

You upgraded successfully.

thank you!

**Reply by JC5** [reply]
Thank YOU! I found the issue and future upgrades should not suffer from the same bug :+1:

**Comment by Zeashh** [comment]
I've been getting this error after upgrades for a while now:

```
Now executing command "correction:restore-oauth-keys"
[2025-10-07 12:06:10] production.ERROR: Could not decrypt pub/private keypair.  
[2025-10-07 12:06:10] production.ERROR: The MAC is invalid.  
  [i] Generated and stored new keys.
```

All sessions are invalid after upgrade, including PATs. The solution above stopped a trend where they would become invalid after restarts, too, though with the newest release I've noticed restarts don't make a difference either way.

If it's somehow relevant, I use a custom image which just adds my locale to the official one.

**Reply by JC5** [reply]
The values collected from the database get invalidated somehow. What database are you running?

**Reply by Zeashh** [reply]
PostgreSQL 15.14

**Reply by JC5** [reply]
Sorry, I have no clue. Do you get this error every time you restart the container(s)?

**Reply by Zeashh** [reply]
Nope, only after upgrades as far as I can tell. Can I safely downgrade from the latest version to test further?

**Reply by JC5** [reply]
It depends a bit, the patch versions should give you no problems.

**Reply by Zeashh** [reply]
You mean something like 6.4.2 -> 6.4.1?

**Reply by JC5** [reply]
Yes

**Reply by Zeashh** [reply]
I've done some testing by switching between the above mentioned versions. The issue appears in downgrades the same way it does in upgrades. I enabled debug logs, but there are no extra logs related to the issue. I thought for a moment it was related to how Kubernetes was rolling out updates by first making the new pod (container) and then deleting the old one, but changing this behavior didn't do anything. Tried doing upgrades/downgrades without any defined oauth tokens but no success.

Frankly I'm not sure what else I can even try.

**Reply by JC5** [reply]
Strange. Thanks for sharing, I will do some testing on my side as well.
